### ISSUE TITLE
Morris - Modify Rule to Not warn on unfilled Lines

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#2053]
_Imported from trac ticket 2053.
Created by **dwarren**
Opened in trac: Dec 19, 2012, 9:40:10 AM
Last modified in trac: Jan 2, 2013, 1:36:00 PM_
***
_by dwarren:_
At Morris,  80% or more of our orders at our Ashley stores cannot be filled at the time of sale.  I am afraid that if associates are seeing the yellow warning on a consistent basis due to inventory not being reserved, they will not pay attention to it when it is for something else.
System Parameter - unfilled warning, default to Enabled for new installations,  Morris will disable.

**Comment by kentbower** [comment]
_**kbower** trac comment on Dec 20, 2012, 10:21:34 AM:_

A rule override mechanism already exists for this; not system parameter, much more flexible since the retailer can selectively display the warning based on the context of the order and order line rather than just on/off.

See kan, eld, isf examples; here only display if the order isn't backorder status (in which case the warning would be expected):

```python
def kan_warn_notfilled(orderdetail):
    return orderdetail.order.deliverydate != DeferredStatus.KAN_BACKORDER
OrderDetail.warn_notfilled = kan_warn_notfilled
```

**Comment by kentbower** [comment]
_**kbower** trac status change on Dec 20, 2012, 10:21:34 AM:_ **closed**

**Comment by kentbower** [comment]
_**sjohnson** trac comment on Dec 20, 2012, 12:14:28 PM:_

I just talked to Mark and Melissa and we all think it should just be turned off.  I think we are leaning towards not filling at the time of sale and allowing our backorder fill process to manage inventory in the near future.  We also don’t want everyone to think there are errors flying all over the place even if it is just notifying of an unfilled line.
 

Darlene

**Comment by kentbower** [comment]
_**sjohnson** trac status change on Dec 20, 2012, 12:14:28 PM:_ **reopened**

**Comment by kentbower** [comment]
_**sjohnson** trac comment on Jan 2, 2013, 1:36:00 PM:_

(Changeset [[3513]](../commit/bedd554d9036fae7a87202f9044f398833af8f6d))
* Fixes #2053 Modify Morris Rule to not warn on unfilled lines

**Comment by kentbower** [comment]
_**sjohnson** trac status change on Jan 2, 2013, 1:36:00 PM:_ **closed**
